cli/azd/cmd/config.go

```diff
@@ -90,10 +90,20 @@ func configActions(root *actions.ActionDescriptor, rootOptions *internal.GlobalC
 		},
 	})
 
+	group.Add("show", &actions.ActionDescriptorOptions{
+		Command: &cobra.Command{
+			Short: "Show all the configuration values.",
+			Long:  `Show all configuration values in ` + userConfigPath + `.`,
+		},
+		ActionResolver: newConfigShowAction,
+		OutputFormats:  []output.Format{output.JsonFormat},
+		DefaultFormat:  output.JsonFormat,
+	})
+
 	group.Add("list", &actions.ActionDescriptorOptions{
 		Command: &cobra.Command{
-			Short: "Lists all configuration values.",
-			Long:  `Lists all configuration values in ` + userConfigPath + `.`,
+			Short:  "List all the configuration values. (Deprecated. Use azd config show)",
+			Hidden: true,
```

Comment:
Thoughts on why we're choosing to hide the command with a warning vs keeping `list` as an alias for `show` till it's fully deprecated.  I don't have a strong opinion here, just wondering.

Reply:
I followed the same strategy from `azd infra create/delete`

I think this strategy allow us to print the `warning` before calling the action.  When using `alias`, I think we would need to make the action's implementation to check if the alias was used to decide when to print the `warning`.